x-pack/plugins/observability_onboarding/public/assets/standalone_agent_setup.sh

```diff
@@ -19,41 +63,40 @@ updateStepProgress() {
     --no-progress-meter
 }
 
-echo "Downloading Elastic Agent archive"
+ELASTIC_AGENT_DOWNLOAD_URL="https://artifacts.elastic.co/downloads/beats/elastic-agent/${artifact}.tar.gz"
+echo "Downloading Elastic Agent archive from ${ELASTIC_AGENT_DOWNLOAD_URL}"
 updateStepProgress "ea-download" "loading"
-ELASTIC_AGENT_DOWNLOAD_URL="https://artifacts.elastic.co/downloads/beats/elastic-agent/elastic-agent-${ELASTIC_AGENT_VERSION}-linux-x86_64.tar.gz"
 curl -L -O $ELASTIC_AGENT_DOWNLOAD_URL --fail --continue-at -
```

Comment:
This fails if the file was fully downloaded already, which can be likely if the script restarted after being killed after download completes.  Reproducible by just running `curl -O https://artifacts.elastic.co/downloads/beats/elastic-agent/elastic-agent-8.8.0-darwin-aarch64.tar.gz -C - --fail` twice in a row (assuming the first succeeds).  This behavior is surprising to me.

A few options we have:
* do nothing.  users will have to delete downloads before trying again
* conditionally resume download.  bake the size of the expected artifact into the script
* remove `--continue-at` option and always download the full artifact
* add another file on disk indicating successful download and skip downloading if it exists

Reply:
I checked another webserver to see how it behaves and it's consistent with artifacts.elastic.co - a range request starting at the length of the artifact returns an HTTP 416 there as well, so I guess this should not be surprising.

Reply:
Mentioned this casually to @axw and he added another option - perform a head request to get the full size first and only continue if the size on disk is smaller.  For the sake of simplicity I've removed the `--continue-at` option for now and we can revisit if desired.